tasks/service_scale.rb

```diff
@@ -6,10 +6,10 @@
 require 'puppet'
 
 def service_scale(service, scale, detach)
-  cmd_string = 'docker service scale'
-  cmd_string += " #{service}" unless service.nil?
-  cmd_string += "=#{scale}" unless scale.nil?
-  cmd_string += ' -d' unless detach.nil?
+  cmd_string = ['docker', 'service', 'scale']
+  cmd_string += [" #{service}"] unless service.nil?
+  cmd_string += ["=#{scale}"] unless scale.nil?
+  cmd_string += [' -d'] unless detach.nil?
```

Comment:
We should try to avoid using += here.

I believe [Array#concat](https://ruby-doc.org/core-2.1.1/Array.html#method-i-concat) would be preferable in this instance.

Also lets rename the variable to cmd rather than cmd_string.

Reply:
In this case you have changed from concatenating a string to an array so we should generally lean towards the more idiomatic approach.